Sources/PackageCollections/PackageCollections.swift

```diff
@@ -258,6 +297,27 @@ public struct PackageCollections: PackageCollectionsProtocol {
         }
     }
 
+    func findPackageByIdentifier(
+        _ identifier: PackageReference.PackageIdentity,
```

Comment:
This might be bikeshedding, but my impression was that the "Swifty" way to name these kinds of things would be `findPackage(identifier: …, profile: …, …)` rather than the nameless first parameter.  The other one in this file would then be `findPackages(query: …, …)`, etc.  Was that considered?